invenio_oauthclient/finalize_app.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2023 Graz University of Technology.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Finalize app."""
+
+from flask import request
+from flask_menu import current_menu
+from invenio_i18n import lazy_gettext as _
+from invenio_theme.proxies import current_theme_icons
+from speaklater import make_lazy_string
+
+
+def finalize_app(app):
+    """Finalize app."""
+    override_template_configuration(app)
+    init_index_menu(app)
+    init_index_breadcrumb()
+    post_ext_init(app)
+
+
+def override_template_configuration(app):
+    """Override template configuration."""
+    template_key = app.config.get(
+        "OAUTHCLIENT_TEMPLATE_KEY",
+        "SECURITY_LOGIN_USER_TEMPLATE",  # default template key
+    )
+    if template_key is not None:
+        template = app.config[template_key]  # keep the old value
+        app.config["OAUTHCLIENT_LOGIN_USER_TEMPLATE_PARENT"] = template
+        app.config[template_key] = app.config.get(
+            "OAUTHCLIENT_LOGIN_USER_TEMPLATE",
+            "invenio_oauthclient/login_user.html",
+        )
+
+
+def init_index_menu(app):
+    """Init index menu."""
+    item = current_menu.submenu("settings.oauthclient")
```

Comment:
Isn't the use of `current_menu` here wrong? Since we're not inside the application context at this point AFAIK.

Wouldn't the alternative be to either call `app.extensions['menu'].submenu(...)` or have the un-initialized `Menu` extension to import from invenio-theme?

Reply:
hm, but the code as it is is working, but it would be more understandable if i we change to `app.extensions['menu'].submenu(...)`

Reply:
both variants `current_menu` and `app.extensions["menu"].submenu(...)` are working as expected. i tested it by changing the text and checking the value on the page (not the best way, but for now i think enough). further i thought that `current_menu` has to work, because otherwise the application would not start.